pkg/node/node.go

```diff
@@ -282,6 +315,41 @@ func (n Node) removeLabel(nodeName string, key string) error {
 	return nil
 }
 
+// removeLabelIfValueMatches will remove a node label given a label key provided the label's value equals matchValue
+func (n Node) removeLabelIfValueMatches(nodeName string, key string, matchValue string) error {
+	type patchRequest struct {
+		Op   string `json:"op"`
+		Path string `json:"path"`
+	}
+
+	var patchReqs []interface{}
+	patchRemove := patchRequest{
+		Op:   "remove",
+		Path: fmt.Sprintf("/metadata/labels/%s", jsonPatchEscape(key)),
+	}
+	payload, err := json.Marshal(append(patchReqs, patchRemove))
+	if err != nil {
+		return fmt.Errorf("An error occurred while marshalling the json to remove a label from the node: %w", err)
+	}
+	node, err := n.fetchKubernetesNode(nodeName)
+	if err != nil {
+		return err
+	}
+	if n.nthConfig.DryRun {
```

Comment:
We need to check whether the value matches (line 343) before doing this. Otherwise, during dry-runs we will always print out "Would have removed label", even when the value doesn't match and thus we would not have removed the label during actual operation.

Can you move this block to right before line 346?

Reply:
Can do!